lib/cambiatus/kyc/kyc_data.ex

```diff
@@ -113,35 +113,150 @@ defmodule Cambiatus.Kyc.KycData do
+          non_null_first_digit: true,
+          dashes_positions: [],
+          string_length: [10]
+        }
 
       "mipyme" ->
-        ~r/^\d-?\d{3}-?\d{6}$/
+        %{
+          regex: ~r/^\d-?\d{3}-?\d{6}$/,
+          non_null_first_digit: false,
+          dashes_positions: [1, 5],
+          string_length: [10]
+        }
 
       "gran_empresa" ->
-        ~r/^\d-?\d{3}-?\d{6}$/
+        %{
+          regex: ~r/^\d-?\d{3}-?\d{6}$/,
+          non_null_first_digit: false,
+          dashes_positions: [1, 5],
+          string_length: [10]
+        }
+    end
+  end
+
+  def document_type_error_handler(changeset, input, opts) do
+    changeset
+    |> check_null_first_digit(input, opts)
+    |> check_dashes_positions(input, opts)
+    |> check_string_length(input, opts)
+    |> check_only_digits(input, opts)
+  end
+
+  # defp check_null_first_digit(changeset, )
+  defp check_null_first_digit(changeset, input, opts) do
+    # changeset =
+    with true <- opts.non_null_first_digit,
+         true <- String.starts_with?(input, "0") do
+      add_error(changeset, :document, "- First digit cannot be zero")
+    else
+      _ ->
+        changeset
+    end
+  end
+
+  defp check_dashes_positions(changeset, input, opts) do
+    # changeset =
+    with positions <- opts.dashes_positions,
+         #  true <- String.length(input) >= Enum.max(positions),
+         graphemes <- String.graphemes(input) do
+      # Check if dashes are in the correct position, if not insert them
+      graphemes =
+        Enum.reduce(positions, graphemes, fn x, acc ->
+          case Enum.at(acc, x) do
+            "-" ->
+              acc
+
+            _ ->
+              List.insert_at(acc, x, "-")
+          end
+        end)
+
+      # Check if the number of dashes is correct
+      if Enum.count(graphemes, &(&1 == "-")) != Enum.count(positions) do
+        add_error(changeset, :document, "- Dashes positions are not valid")
+      else
+        changeset
+      end
+    else
+      _ ->
+        changeset
+    end
+  end
+
+  defp check_string_length(changeset, input, opts) do
+    input_length =
+      input
+      |> String.replace("-", "")
+      |> String.length()
+
+    # changeset =
+    with pattern_length <- opts.string_length,
+         min_length <- Enum.min(pattern_length),
+         max_length <- Enum.max(pattern_length),
+         true <- input_length < min_length || input_length > max_length do
+      if min_length == max_length do
+        add_error(
+          changeset,
+          :document,
+          "- Entry must be #{min_length} digits long"
+        )
+      else
+        add_error(
+          changeset,
+          :document,
+          "- Entry must be between #{min_length} and #{max_length} digits long"
+        )
+      end
+    else
+      _ ->
+        changeset
+    end
+  end
+
+  defp check_only_digits(changeset, input, _opts) do
+    input = String.replace(input, "-", "")
+
+    # changeset =
```

Comment:
mais um comentário perdido. Se te ajudar você pode colocar um 
```suggestion
    # TODO: remover
    # changeset = 
```

Esse `TODO:` é uma sintaxe especial que é compatível com várias ferramentas de desenvolvimento, inclusive com as padrão que vem no Visual Studio, que te ajudam a lembrar de pedaços de código como esse que ficaram pra trás na revisão

Reply:
Valeu pela dica! Esses comentários passaram batidos por mim, foi mal pelo trabalho a mais.